module/epochs/staker.go

```diff
@@ -0,0 +1,41 @@
+package epochs
+
+import (
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/state/protocol"
+	"github.com/onflow/flow-go/utils/logging"
+)
+
+type Staker interface {
+	AmIStakedAt(blockID flow.Identifier) bool
+}
+
+type SealedStateStaker struct {
+	state protocol.State
+	local module.Local
+	log   zerolog.Logger
+}
+
+func NewSealedStateStaker(log zerolog.Logger, state protocol.State, local module.Local) *SealedStateStaker {
+	return &SealedStateStaker{
+		state: state,
+		local: local,
+		log:   log,
+	}
+}
+
+func (s *SealedStateStaker) AmIStakedAt(blockID flow.Identifier) bool {
+
+	identity, err := s.state.AtBlockID(blockID).Identity(s.local.NodeID())
+	if err != nil {
+		s.log.Warn().Err(err).Hex("block_id", logging.ID(blockID)).Msgf("could not retrieve my own identity while checking for being staked")
```

Comment:
would we hit unknown block?

Reply:
it can come from input, so its possible

Reply:
if the block is unknown, we can't just return `false` and say it's not staked, right. It could be staked, we just don't know the block yet. 

If we simply use a `IsNodeStakedAtBlockID` function (instead of the staker), then we don't have this problem, because at the point we call this function, we have the context to know whether the block would be unknown or not.

Reply:
But it would bring back the conversation how to mock it easily in tests https://github.com/onflow/flow-go/pull/400#discussion_r585022381

Reply:
> if the block is unknown, we can't just return false and say it's not staked, right. It could be staked, we just don't know the block yet.

That's the purpose of having separate `Staker` component - it decides at the moment, and always return true or false. If we don't know the block we aren't staked for it at the moment. 

> `IsNodeStakedAtBlockID` 

This function has been removed, and I literally used its code here (with different error handling).